hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-common/src/test/java/org/apache/hadoop/yarn/server/uam/TestUnmanagedApplicationManager.java

```diff
@@ -518,6 +518,10 @@ public void testSeparateThreadWithoutBlockServiceStop() throws Exception {
     Assert.assertNotNull(unmanagedAppMasterMap);
     Assert.assertEquals(2, unmanagedAppMasterMap.size());
 
+    // threadpool may be shutdown before finishApplicationThread is executed.
+    threadpool.shutdownNow();
```

Comment:
In actual situations, what situations would cause the thread pool to be terminated?